test/Spec.hs

```diff
@@ -322,3 +330,195 @@ integrationTests = testGroup "integrationTests" $
+        , iweight  = 300
+        , idamage  = 1 `d` 1
+        , iaspects = [ SetFlag Ability.Equipable ] 
+        , ieffects = []
+        , idesc    = "A lump of brittle sandstone rock."
+        , ikit     = []
+        }
+      @?= []
+  , testCase "equipableNoSlot\"Symbol_validateSingle_noErr" $
+      validateSingle Content.RuleKind.standardRules ItemKind
+        { isymbol  = '"'
+        , iname    = "12345678901234567890123"
+        , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]  
+        , iflavour = zipPlain [Green]
+        , icount   = 1 + 1 `d` 2 
+        , irarity  = [(1, 50), (10, 1)]
+        , iverbHit = "hit"
+        , iweight  = 300
+        , idamage  = 1 `d` 1
+        , iaspects = [ SetFlag Ability.Equipable ] 
+        , ieffects = []
+        , idesc    = "A lump of brittle sandstone rock."
+        , ikit     = []
+        }
+      @?= []
+  , testCase "equipableNoSlot-Symbol_validateSingle_noErr" $
+      validateSingle Content.RuleKind.standardRules ItemKind
+        { isymbol  = '-'
+        , iname    = "12345678901234567890123"
+        , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]  
+        , iflavour = zipPlain [Green]
+        , icount   = 1 + 1 `d` 2 
+        , irarity  = [(1, 50), (10, 1)]
+        , iverbHit = "hit"
+        , iweight  = 300
+        , idamage  = 1 `d` 1
+        , iaspects = [ SetFlag Ability.Equipable ] 
+        , ieffects = []
+        , idesc    = "A lump of brittle sandstone rock."
+        , ikit     = []
+        }
+      @?= []   
+  , testCase "equipableNoSlot*CustomRules_validateSingle_noErr" $
+      validateSingle customRules ItemKind
+            { isymbol  = '*'
+            , iname    = "12345678901234567890123"
+            , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]  
+            , iflavour = zipPlain [Green]
+            , icount   = 1 + 1 `d` 2 
+            , irarity  = [(1, 50), (10, 1)]
+            , iverbHit = "hit"
+            , iweight  = 300
+            , idamage  = 1 `d` 1
+            , iaspects = [ SetFlag Ability.Equipable ] 
+            , ieffects = []
+            , idesc    = "A lump of brittle sandstone rock."
+            , ikit     = []
+            }
+        @?= []  
+  , testCase "equipableNoSlot\"CustomRules_validateSingle_errs" $                
+      validateSingle customRules ItemKind
+        { isymbol  = '"'
+        , iname    = "12345678901234567890123"
+        , ifreq    = [ (UNREPORTED_INVENTORY, 1) ]  
+        , iflavour = zipPlain [Green]
+        , icount   = 1 + 1 `d` 2 
+        , irarity  = [(1, 50), (10, 1)]
+        , iverbHit = "hit"
+        , iweight  = 300
+        , idamage  = 1 `d` 1
+        , iaspects = [ SetFlag Ability.Equipable ] 
+        , ieffects = []
+        , idesc    = "A lump of brittle sandstone rock."
+        , ikit     = []
+        }
+      @?= ["EqpSlot not specified but Equipable or Meleeable and not a likely organ or necklace or template"]
+  ]
+
+pointUnitTests :: TestTree
```

Comment:
Amazing, thank you. I'd even point here from the definition of `chessDist` so that people unfamiliar with the metric can quickly build intuition. Hmm, actually, even better would be writing this test as https://hackage.haskell.org/package/doctest as part of the haddock comment for `chessDist`, as in 
```
-- >>> chessDist (Point 0 0) (Point 0 0)
-- 0
-- >>> chessDist (Point (-1) (-1)) (Point 1 1)
-- 2
chessDist :: ...
```
and have it automatically executed. It's not set up for LambdaHack, but definitely worth a PR later on.

Reply:
Oh, that's rad. I've never heard of doctest before. Now I want it for c++.

Reply:
Right. It's displayed both in haddocs on Hackage, present in source code adjacent to your function and it's tested. The only problem is the test can't be too long and tedious so that the haddocks don't get boring. :)